internal/api/request.go

```diff
@@ -146,6 +146,12 @@ func (client *Client) buildCorrelationHeaders(ctx context.Context) (sessionId st
 func (client *Client) buildUserAgent(ctx context.Context) string {
 	userAgent := fmt.Sprintf("terraform-provider-power-platform/%s (%s; %s) terraform/%s go/%s", common.ProviderVersion, runtime.GOOS, runtime.GOARCH, client.Config.TerraformVersion, runtime.Version())
 
+	if client.Config.PartnerId != "" {
```

Comment:
can we get that refactored to a separate private function? `configurePartnerId(...)

Reply:
i don't think it adds clarity to be in a different function when all the other string concat is in this function